File: 36. Valid Sudoku/36. Valid Sudoku.cs
public class Solution {
    public bool IsValidSudoku(char[][] board) {
        bool[,] rows = new bool[9, 9];
        bool[,] cols = new bool[9, 9];
        bool[,] boxes = new bool[9, 9];
        
        for (int i = 0; i < 9; i++) {
            for (int j = 0; j < 9; j++) {
                if (board[i][j] != '.') {
                    int num = board[i][j] - '1';
                    int boxIndex = (i / 3) * 3 + j / 3;
                    
                    if (rows[i, num] || cols[j, num] || boxes[boxIndex, num]) {
                        return false;
                    }
                    
                    rows[i, num] = cols[j, num] = boxes[boxIndex, num] = true;
                }
            }
        }
        
        return true;
    }
}

File: 36. Valid Sudoku/36. Valid Sudoku.rs
impl Solution {
    pub fn is_valid_sudoku(board: Vec<Vec<char>>) -> bool {
        let mut rows = [[false; 9]; 9];
        let mut cols = [[false; 9]; 9];
        let mut boxes = [[false; 9]; 9];
        
        for i in 0..9 {
            for j in 0..9 {
                if board[i][j] != '.' {
                    let num = (board[i][j] as u8 - b'1') as usize;
                    let box_index = (i / 3) * 3 + j / 3;
                    
                    if rows[i][num] || cols[j][num] || boxes[box_index][num] {
                        return false;
                    }
                    
                    rows[i][num] = true;
                    cols[j][num] = true;
                    boxes[box_index][num] = true;
                }
            }
        }
        
        true
    }
}